packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/TableDemo/TableEditableDemo.tsx

```diff
@@ -0,0 +1,304 @@
+import * as React from 'react';
+import { TextInput } from '@patternfly/react-core';
+import {
+  Table,
+  TableHeader,
+  TableBody,
+  TableProps,
+  TableVariant,
+  ICell,
+  IRow,
+  IRowCell,
+  IValidatorDef,
+  OnRowEdit,
+  validateCellEdits,
+  cancelCellEdits,
+  applyCellEdits
+} from '@patternfly/react-table';
+
+const rowLevelValidationRules: IValidatorDef[] = [
+  {
+    name: 'required',
+    validator: value => value.trim() !== '',
+    errorText: 'This field is required'
+  },
+  {
+    name: 'notFoo',
+    validator: value => value.trim().toLowerCase() !== 'foo',
+    errorText: 'Value cannot be "foo"'
+  },
+  {
+    name: 'minLength',
+    validator: value => value.trim().length >= 7,
+    errorText: 'Value must be at least 7 characters'
+  },
+  {
+    name: 'notXyz',
+    validator: value => value.trim().toLowerCase() !== 'xyz',
+    errorText: 'Value cannot be xyz'
+  }
+];
+
+export class TableEditableDemo extends React.Component<TableProps, { columns: (ICell | string)[]; rows: IRow[] }> {
+  constructor(props: TableProps) {
+    super(props);
+    this.state = {
+      columns: [
+        'Text input col 1',
+        'Disabled text input col 2',
+        'Text input col 3',
+        'Text input col 4'
+      ],
+      rows: [
+        {
+          isEditable: true,
+          rowEditValidationRules: rowLevelValidationRules,
+          cells: [
+            {
+              title: (value: string, rowIndex: number, cellIndex: number, props: any) => (
+                <React.Fragment>
+                  <div className="pf-c-inline-edit__value">{value}</div>
+                  <div className="pf-c-inline-edit__input">
+                    <TextInput
+                      value={props.editableValue !== undefined ? props.editableValue : value}
+                      isValid={props.isValid}
+                      type="text"
+                      onChange={(value, event) => {
+                        this.handleTextInputChange(value, event, rowIndex, cellIndex);
+                      }}
+                      aria-label="Row 1 cell 1 content" />
+                    <div className="pf-c-form__helper-text pf-m-error" aria-live="polite">{props.errorText}</div>
+                  </div>
```

Comment:
+1. I'd like to see another TextInupt variation, specifically for inline edit, rather than manually wrap component with pf-c-inline-edit__input

Reply:
I asked the core folks about creating a dedicated variation of TextInput for inline edit, but it didn't seem desirable. When you initially left this comment, I hadn't gotten a chance to make use of that `EditableTextCell` wrapper component I created yet - just added it, thinking it will help ease the concern here. Let me know if you see any issues with this approach? Thx!

Reply:
Yes, I like this better